autobahn/wamp/interfaces.py

```diff
@@ -647,3 +667,70 @@ class IAuthenticator(object):
     def on_challenge(session, challenge):
         """
         """
+
+
+@public
+@six.add_metaclass(abc.ABCMeta)
+class IPayloadCodec(object):
+    """
+    WAMP payload codecs are used with WAMP payload transparency mode.
+
+    In payload transparency mode, application payloads are transmitted "raw",
+    as binary strings, without any processing at the WAMP router.
+
+    Payload transparency can be used eg for these use cases:
+
+    * end-to-end encryption of application payloads (WAMP-cryptobox)
+    * using serializers with custom user types, where the serializer and
+      the serializer implementation has native support for serializing
+      custom types (such as CBOR)
+    * transmitting MQTT payloads within WAMP, when the WAMP router is
+      providing a MQTT-WAMP bridge
+    """
+
+    @public
+    @abc.abstractmethod
+    def encode(is_originating, uri, args=None, kwargs=None):
+        """
+        Encodes application payload.
+
+        :param is_originating: Flag indicating whether the encoding
+            is to be done from an originator (a caller or publisher).
+        :type is_originating: bool
+
```

Comment:
I recently learned a fun trick: the above can be `:param bool is_originating` and therefore omit the "type" thing entirely.